### ISSUE TITLE
GeolocateControl wrong pixel offset

### ISSUE THREAD

**Issue by davidnx** [visgl/react-map-gl#1067]
The user's position is rendered off by a few pixels because it doesn't set Marker's offsetTop and offsetLeft. The marker has size 15x15px by default, and without the offset, the center of the circle is not where the user actualy is. This leads to gross position errors, especially at low zoom levels.

I see two trivial possible fixes:

1. Add props òffsetTop={-7.5} offsetLeft={-7.5}` in [geolocate-control.js:260](https://github.com/uber/react-map-gl/blob/6fa913270c7e074ba903e9cfcae51c0572aaf687/src/components/geolocate-control.js#L260)

2. Change the `Marker` component so that it doesn't set `top: 0, left: 0` in inline style. That way, user CSS could specify `left: -50%; right: -50%;` to get the right alignment.

IMHO (2) is better, as it avoids hard-coded constants outside of CSS.

**Comment by alexnorton** [comment]
FYI I have temporarily fixed this in my app by adding the following CSS:

```css
.mapboxgl-user-location-dot {
  left: -100% !important;
  top: -100% !important;
}
```

**Comment by Pessimistress** [comment]
Fixed in 5.2.10
